src/header.rs

```diff
@@ -47,67 +44,20 @@ impl Header {
     pub fn from_reader(metadata: &std::fs::Metadata, mut rdr: impl Read) -> io::Result<Self> {
         let file_sz = metadata.len();
 
-        let magic = rdr.read_u32::<NativeEndian>()?;
-
-        // determine db file version, intrinsics from magic number
-        let (is_lfs, need_swap) = match magic {
-            GDBM_OMAGIC => (false, false),
-            GDBM_OMAGIC_SWAP => (false, true),
-            GDBM_MAGIC32 => (false, false),
-            GDBM_MAGIC32_SWAP => (false, true),
-            GDBM_MAGIC64 => (true, false),
-            GDBM_MAGIC64_SWAP => (true, true),
-            _ => {
-                return Err(Error::new(ErrorKind::Other, "Unknown/invalid magic number"));
-            }
-        };
-
-        // detect db file endianness
-        let is_le = match need_swap {
-            true => !cfg!(target_endian = "little"),
-            false => cfg!(target_endian = "little"),
-        };
+        let magic = Magic::from_reader(&mut rdr)?;
 
-        // fixme: read u32, not u64, if is_lfs
```

Comment:
You dropped the fixme without fixing the highlighted issue :)

the u64's are really offsets.

plus, again, this introduces the complication of padding.